src/ansys/dpf/core/data_tree.py

```diff
@@ -482,6 +482,7 @@ def get_as(self, name, type_to_return=types.string):
             out = DataTree(data_tree=obj, server=self._server)
         return out
 
+    @property
     def get_attribute_names(self):
```

Comment:
Also above, why not just call the methods `to_json` or `from_txt` without the `read` and `write` parts. This too seems explicit enough without it.

Reply:
@jose-henriquezroa you do not have to implement the setter and make it raise an error if what you want is for people to not be able to set. Simply not specifying a setter is enough.

Reply:
@PProfizi You're right, my mistake. I misunderstood  the "property". 

Nonetheless, in the very particular case of the DataTree we've also defined the __setattr__ function. Which means that if the user does data_tree.get_attribute_name = "hello", instead of throwing an error it's going to define an attribute in the data_tree object with key "get_attribute_name" and value "hello". So I guess using "properties" is too risky here still.